fred2/eventeditor.cpp

```diff
@@ -796,9 +796,23 @@ void event_editor::OnCancel()
 // this is called the clicking the ID_CANCEL button
 void event_editor::On_Cancel()
 {
+	int z;
+
 	audiostream_close_file(m_wave_id, 0);
 	m_wave_id = -1;
 
+	if (query_modified()) {
+		z = MessageBox("Do you want to keep your changes?", "Close", MB_ICONQUESTION | MB_YESNOCANCEL);
```

Comment:
1. `z` is not a very descriptive variable name. Maybe call it something like `choice`?
2. The scope of variables should be as small as possible so you should declare the variable when you need it first and not before. I know that most of FRED is written with most variables declared at the beginning of the function but that code is old and uses an outdated coding style.
3. Maybe use a `switch`-`case` here? I think that would make the code a little easier to read.